### PR TITLE
Change cson library

### PR DESCRIPTION
**Pull request by ericadamski**
robustified object merging

### CODE REVIEW

**Comment by ericadamski** [comment]
@pvienneau can you checkout this branch and run `apm link -d; atom -d .` and then `ctrl-alt-cmd-s` to sync your snippets and see if it works? Also, you should backup the snippets aswell

File: lib/manager.js [hunk 1]
```diff
@@ -28,64 +29,84 @@ export default class Manager {
...
+                return JSON.parse(fs.readFileSync(path.join(TARGET, file), 'utf8'));
         });
     }
 
+    merge(destination, ...sources) {
+        sources.map(source => {
+                keys(source).map(key => {
+                        console.log(sources);
+                        console.log(source);
+                        console.log(key);
```
Comment (korygorsky) [inline lib/manager.js]: Did you want to remove these console logs?

Comment (ericadamski) [inline lib/manager.js]: done.

File: lib/manager.js [hunk 2]
```diff
@@ -28,64 +28,81 @@ export default class Manager {
...
+        try { fs.statSync(TARGET); }
+        catch (e) { mkdirp.sync(TARGET); }
 
         return this.files.then(files =>
             Promise.all(files.map(file => {
-                return new Promise(resolve =>
-                    fs.symlink(file, path.join(TARGET, path.basename(file)), fid => resolve(fid)));
+                return new Promise(resolve => {
+                    try {
+                        fs.symlink(file, path.join(TARGET, path.basename(file)), fid => resolve(fid))
```
Comment (korygorsky) [inline lib/manager.js]: Should this have an ending comma?

Comment (ericadamski) [inline lib/manager.js]: No, it is a function call. Trailing commas are for `{}` and `[]`

Comment (korygorsky) [inline lib/manager.js]: And by comma, I mean semi-colon...

**Pull request merged**